Only netstandard projects

masojus opened the pull request:
Remove the 3.5/4.5/PCL projects, so now there's only the netstandard2.0/netfx4.5 multi-targeted project and the netcore2.0 test project.

Lots of moved files, so the interesting parts are the .sln/.csproj and config/script/props files related to building/packing/CI/testing/etc.

masojus committed:
Remove the 3.5/4.5/PCL projects from the solution and get rid of the build/solution configs that were for netfx/netstandard subsets of those projects.

masojus committed:
Remove the actual files in the old projects.

masojus committed:
Remove stale NUnit project file. We can add this back and update it if it seems useful for CI builds later.

masojus committed:
Bump some of the test dependency versions to use the latest and greatest.

masojus committed:
Try out CI with only the NetStandard projects.

masojus committed:
First try at updating the packaging script to only use the netstandard projects and build the new .nupkg.

masojus committed:
There seems to be an issue with Moq pdbs missing causing OpenCover to report 0%, so try filtering those namespaces.

masojus committed:
Minor comment about AppVeyor.

masojus committed:
Merge remote-tracking branch 'github/master' into jm_OnlyNetStandardProj

masojus commented on NewSemVer.ps1:
```diff
@@ -24,9 +24,8 @@ $semVerPattern = '\d+.\d+.\d+(-[a-zA-Z0-9-]+)?'
-    "the SharedVersionInfo.cs file. Then, tries to create a new NuGet package, which will fail if "
-    "nuget.exe isn't in the PATH or the script's directory.`n"
+    "Updates the Version in the .csproj file. Then, tries to create a new NuGet package, which will"
+    "fail if nuget.exe isn't in the PATH or the script's directory.`n"
```
This comment is wrong, since we aren't directly using `nuget.exe` but rather rely on the dotnet core tools being installed, so I'll need to fix that.

masojus replied:
Fixed in another commit.

baumatron commented on .appveyor.yml:
```diff
@@ -102,7 +96,7 @@ test_script:
-  - .\tools\OpenCover.4.6.519\tools\OpenCover.Console.exe -target:dotnet.exe -targetargs:"test --logger \"trx;LogFileName=netstandard_test_results.xml\" .\Keen.NetStandard.Test\Keen.NetStandard.Test.csproj /property:DebugType=Full" -mergeoutput -output:coverage.xml -filter:"+[*]* -[*.Test]*" -oldStyle -returntargetcode -register:user
+  - .\tools\OpenCover.4.6.519\tools\OpenCover.Console.exe -target:dotnet.exe -targetargs:"test --logger \"trx;LogFileName=netstandard_test_results.xml\" .\Keen.NetStandard.Test\Keen.NetStandard.Test.csproj /property:DebugType=Full" -mergeoutput -output:coverage.xml -filter:"+[*]* -[*.Test]* -[Moq*]*" -oldStyle -returntargetcode -register:user
 
   # Upload merged coverage results from both test runs
```
This comment is outdated.

baumatron commented on .appveyor.yml:
```diff
@@ -102,7 +96,7 @@ test_script:
-  - .\tools\OpenCover.4.6.519\tools\OpenCover.Console.exe -target:dotnet.exe -targetargs:"test --logger \"trx;LogFileName=netstandard_test_results.xml\" .\Keen.NetStandard.Test\Keen.NetStandard.Test.csproj /property:DebugType=Full" -mergeoutput -output:coverage.xml -filter:"+[*]* -[*.Test]*" -oldStyle -returntargetcode -register:user
+  - .\tools\OpenCover.4.6.519\tools\OpenCover.Console.exe -target:dotnet.exe -targetargs:"test --logger \"trx;LogFileName=netstandard_test_results.xml\" .\Keen.NetStandard.Test\Keen.NetStandard.Test.csproj /property:DebugType=Full" -mergeoutput -output:coverage.xml -filter:"+[*]* -[*.Test]* -[Moq*]*" -oldStyle -returntargetcode -register:user
 
   # Upload merged coverage results from both test runs
```
This comment is outdated.

masojus replied:
Ha, I was looking at OpenCover [Issue 601](https://github.com/OpenCover/opencover/issues/601) thinking you meant the `-oldStyle` flag wasn't needed anymore, but then I realized you were referring to the "from both test runs" part :satisfied:, will fix.

baumatron reviewed (approved):
Awesome! Good to see these gone.

masojus committed:
PR Feedback: fix a couple comments in config/scripts that still refer to old solution structure.

masojus commented:
Merging even though the MacOS CI job hasn't finished...changes since last successful MacOS build are comment-only. If there's an issue, I'll deal with the merge of this and PR #136 directly in `master`, but it'd unlikely.
